src/main/java/org/jenkinsci/plugins/microsoft/appservice/commands/DefaultDockerClientBuilder.java

```diff
@@ -27,6 +30,12 @@ public DockerClient build(AuthConfig authConfig) {
                 .withRegistryUrl(authConfig.getRegistryAddress())
                 .withRegistryEmail(authConfig.getEmail());
 
+        // must enable tcp on windows by check the option "Expose daemon on tcp://localhost:2375 without TLS"
+        // more reading at https://docs.microsoft.com/en-us/virtualization/windowscontainers/manage-docker/configure-docker-daemon
+        if (SystemUtils.IS_OS_WINDOWS) {
```

Comment:
It happens while trying to get a `DockerClient`. This method itself is neutral.  It's run on master if it's caller executed on master. Same to slave.
I think only the validation in config phase is run on master,  while docker commands in job runs should be run in slaves.